### PR TITLE
RavenDB-17617 Disable creating auto indexes via studio queries - Stud…

### PR DESCRIPTION
**Pull request by Danielle9897**
…io part only

### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-17617

### Additional description
Allow querying from Studio without creating a new auto-index.
This PR is **Studio work only**.
See detailed description in:
https://issues.hibernatingrhinos.com/issue/RavenDB-17617#focus=Comments-67-270121.0-0

### Type of change
- New feature

### How risky is the change?
- Low 

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- This change requires a documentation update. 

### Testing 
- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### Server work
- Further Server work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts [hunk 1]
```diff
@@ -205,7 +208,16 @@ class patch extends viewModelBase {
 
         this.loadLastQuery();
 
-        return $.when<any>(this.fetchAllIndexes(this.activeDatabase()), this.savedPatches.loadAll(this.activeDatabase()));
+        const studioDatabaseConfigTask = new getDatabaseStudioConfigurationCommand(this.activeDatabase())
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts]: IIRC studio confgiratuoin is loaded when we change db and it is persistent in memory. Please have a look at usages.

File: src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts [hunk 2]
```diff
@@ -205,7 +208,16 @@ class patch extends viewModelBase {
...
+            .execute()
+            .done((settings: Raven.Client.Documents.Operations.Configuration.StudioConfiguration) => {
+                const test = settings.DisableAutoIndexCreation;
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts]: rename variable

File: src/Raven.Studio/typescript/viewmodels/database/query/query.ts [hunk 3]
```diff
@@ -578,8 +581,16 @@ class query extends viewModelBase {
         
         const db = this.activeDatabase();
 
-        return this.fetchAllIndexes(db)
+        const studioDatabaseConfigTask = new getDatabaseStudioConfigurationCommand(this.activeDatabase())
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: same here.

File: src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/query/query.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: done

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs [hunk 4]
```diff
@@ -119,6 +119,9 @@ private async Task Query(QueryOperationContext queryContext, OperationCancelToke
             
             indexQuery.Diagnostics = diagnostics ? new List<string>() : null;
             indexQuery.AddTimeSeriesNames = GetBoolValueQueryString("addTimeSeriesNames", false) ?? false;
+            
+            // TODO for: RavenDB-17617
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: what is the todo for?

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs [hunk 5]
```diff
@@ -476,7 +479,10 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
         public async Task Patch()
         {
             var queryContext = QueryOperationContext.Allocate(Database); // we don't dispose this as operation is async
-
+            
+            // TODO for: RavenDB-17617
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: ??

File: src/Raven.Studio/typescript/viewmodels/database/query/query.ts [hunk 6]
```diff
@@ -45,6 +45,8 @@ import rqlLanguageService = require("common/rqlLanguageService");
 import hyperlinkColumn = require("widgets/virtualGrid/columns/hyperlinkColumn");
 import moment = require("moment");
 import { highlight, languages } from "prismjs";
+import getDatabaseStudioConfigurationCommand = require("commands/resources/getDatabaseStudioConfigurationCommand");
+import activeDatabaseTracker from "common/shell/activeDatabaseTracker";
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: require?

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: done

**Review by ml054 — APPROVED** [review]

**Comment by arekpalinski** [comment]
The PR is ready to review / merge

File: src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs [hunk 7]
```diff
@@ -153,7 +153,10 @@ public async Task<(Index Index, bool HasCreatedAutoIndex)> CreateAutoIndexIfNeed
             while (TryMatchExistingIndexToQuery(map, out index) == false)
             {
                 if (createAutoIndexIfNoMatchIsFound == false)
-                    throw new IndexDoesNotExistException("Could not find index for a given query.");
+                    throw new IndexDoesNotExistException("No Auto Index matching the given query was found.");
+
+                if (query.DisableAutoIndexCreation)
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs:158]: shouldn't we use `createAutoIndexIfNoMatchIsFound` here?

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs (hunk 4 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: It was TODO that Danielle added to handle server side work. It was removed in later commit

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs (hunk 5 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: It was TODO that Danielle added to handle server side work. It was removed in later commit

File: src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs (hunk 7 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs:158]: No, `createAutoIndexIfNoMatchIsFound ` has different purpose. We want to throw a different exception with better error message (`IndexDoesNotExistException` is translated to `404` status code later on - so it just shows `Index not found` in the studio)

**Pull request merged**